include/xtensor/xutils.hpp

```diff
@@ -857,6 +857,19 @@ namespace xt
 
     template <class ST>
     using inner_reference_t = typename inner_reference<ST>::type;
+
+    /************
+     * has_rank *
+     ************/
+
+    template <class E, size_t N>
+    struct has_rank
+    {
+        using type = std::integral_constant<bool, E::rank == N>;
```

Comment:
You should replace `E::rank` with `std::decay_t<E>::rank` so it can work when E is a reference.

Reply:
It always gets me... Thanks